Using AOT with clojurescript dependency throws ClassCastException

Hi @ztellman just updating this problem in a clean and separate issue 

### Description 
+ adding `org.clojure/clojurescript` dependency and `:aot :all` with `ztellman/automat` throws error when executing standalone jar
` Caused by: java.lang.ClassCastException: clojure.lang.Var$Unbound cannot be cast to clojure.lang.MultiFn`

### example code
https://github.com/tangrammer/automat-issue

--- a/src/automat/fsm.cljc
+++ b/src/automat/fsm.cljc
@@ -868,11 +868,8 @@
               ::none)))))))
 
 ; cljs.compiler tries to emit metadata on precompiled automata and chokes on IAutomaton
-#?(:clj (do
-          (defmacro ^:private cljs-emit-constant []
-            (when (try
-                    (-> 'cljs.compiler require nil?)
-                    (catch Exception _))
-              `(defmethod cljs.compiler/emit-constant ~IAutomaton [x#]
-                 (cljs.compiler/emit-constant nil))))
-          (cljs-emit-constant)))
+#?(:clj (defmacro ^:private cljs-emit-constant []
+          (eval '(defmethod cljs.compiler/emit-constant IAutomaton [x]
+                   (cljs.compiler/emit-constant nil)))
+          nil)
+   :cljs (automat.fsm/cljs-emit-constant))

